components/progress-rope/src/ProgressRope.js

```diff
@@ -0,0 +1,103 @@
+/** @jsx jsx */
+
+import { Children, cloneElement, createContext, useState, useEffect, useContext } from 'react';
+import PropTypes from 'prop-types';
+import { jsx } from '@westpac/core';
+import { ProgressRopeGroup } from './ProgressRopeGroup';
+
+// ==============================
+// Context and Consumer Hook
+// ==============================
+const ProgressRopeContext = createContext();
+
+export const useProgressRopeContext = () => {
+	const context = useContext(ProgressRopeContext);
+
+	if (!context) {
+		throw new Error('ProgressRope sub-components should be wrapped in a <ProgressRope>.');
+	}
+
+	return context;
+};
+
+// ==============================
+// Utils
+// ==============================
+const createRopeGraph = children => {
+	const ropeGraph = [];
+	let grouped = false;
+	Children.forEach(children, (child, i) => {
+		if (child.type === ProgressRopeGroup) {
+			grouped = true;
+			ropeGraph.push(Array(Children.count(child.props.children)).fill('unvisited'));
+		} else {
+			ropeGraph.push(['unvisited']);
+		}
+	});
+	return { ropeGraph, grouped };
+};
+
+// ==============================
+// Component
+// ==============================
+export const ProgressRope = ({ current, children, ...props }) => {
+	const [progress, setProgress] = useState({
+		currStep: current,
+		currGroup: 0,
+		openGroup: 0,
+		...createRopeGraph(children),
+	});
+
+	const handleClick = index => {
+		setProgress({ ...progress, openGroup: index !== progress.openGroup ? index : null });
```

Comment:
This would come up as stale state here. Easier to separate them into each states or `useReducer`